src/frontend/js/hooks/useResources/useResourcesRoot.ts

```diff
@@ -0,0 +1,182 @@
+  },
+  errorNotFound: {
+    id: 'hooks.useResources.errorNotFound',
+    description: 'Error message shown to the user when no resources matches.',
+    defaultMessage: 'Cannot find the resource',
+  },
+  errorUpdate: {
+    id: 'hooks.useResources.errorUpdate',
+    description: 'Error message shown to the user when resource update request fails.',
+    defaultMessage: 'An error occurred while updating a resource. Please retry later.',
+  },
+  errorDelete: {
+    id: 'hooks.useResources.errorDelete',
+    description: 'Error message shown to the user when resource deletion request fails.',
+    defaultMessage: 'An error occurred while deleting a resource. Please retry later.',
+  },
+  errorCreate: {
+    id: 'hooks.useResources.errorCreate',
+    description: 'Error message shown to the user when resource creation request fails.',
+    defaultMessage: 'An error occurred while creating a resource. Please retry later.',
+  },
+});
+
+type MutateFunc<TApiMethod extends Maybe<(...args: any[]) => any>> = AddParameters<
+  NonNullable<TApiMethod>,
+  [options?: MutateOptions<Awaited<ReturnType<NonNullable<TApiMethod>>>, HttpError>]
+>;
+
+export const useResourcesRoot = <
+  TData extends Resource,
+  TResourceQuery extends ResourcesQuery = ResourcesQuery,
+  TApiResource extends ApiResourceInterface<TData> = ApiResourceInterface<TData>,
+>({
+  queryKey,
+  apiInterface,
+  frozenQueryKey,
+  filters,
+  session,
+  queryOptions,
+  localized,
+  messages: resourceMessages,
+}: UseResourcesProps<TData, TResourceQuery, TApiResource>) => {
+  const queryClient = useQueryClient();
+
+  const [error, setError] = useState<Maybe<string>>();
+  const intl = useIntl();
+  const COMMON_QUERY_KEY = frozenQueryKey ? [...queryKey] : [...queryKey, JSON.stringify(filters)];
+  const LOCALIZED_QUERY_KEY = useLocalizedQueryKey(COMMON_QUERY_KEY);
+  const QUERY_KEY = localized ? LOCALIZED_QUERY_KEY : COMMON_QUERY_KEY;
+  let ACTUAL_QUERY_KEY = QUERY_KEY;
+  const actualMessages = { ...messages, ...resourceMessages };
+  const api = apiInterface();
+
+  const queryFn: () => Promise<any> = () => api.get(filters);
+  queryOptions = {
+    onError: () => setError(intl.formatMessage(actualMessages.errorSelect)),
+    ...queryOptions,
+  };
+
+  if (session !== usePrevious(session)) {
+    throw new Error('session must never change value.');
+  }
+
+  let readHandler: UseQueryResult<any, HttpError>;
+  if (session) {
+    [readHandler, ACTUAL_QUERY_KEY] = useSessionQuery(QUERY_KEY, queryFn, queryOptions as any);
+  } else {
+    readHandler = useQuery(QUERY_KEY, queryFn, queryOptions);
+  }
+
+  const invalidate = async () => {
+    // Invalidate all queries related to the resource
+    await queryClient.invalidateQueries({
+      predicate: (query) => query.queryKey.includes(queryKey[0]),
+    });
+  };
+
+  const prefetch = async () => {
+    await queryClient.prefetchQuery(ACTUAL_QUERY_KEY, queryFn, {
+      staleTime: REACT_QUERY_SETTINGS.staleTimes.sessionItems,
```

Comment:
`staleTime` should be aware if the resource relies on the session or not.

```suggestion
      staleTime: session ? REACT_QUERY_SETTINGS.staleTimes.sessionItems : REACT_QUERY_SETTINGS.default,
```